erroranalysis/erroranalysis/_internal/matrix_filter.py

```diff
@@ -292,6 +376,24 @@ def matrix_2d(categories1, categories2, matrix_counts,
                 false_count = 0
                 if index_exists_err and col_exists_err:
                     false_count = int(matrix_err_counts.loc[cat1, cat2])
+            elif is_precision:
+                tp_sum = []
+                pred_sum = []
+                metric_value = 0
+                if index_exists_err and col_exists_err:
+                    metric_value = float(matrix_err_counts.loc[cat1, cat2][0])
+                    tp_sum = matrix_err_counts.loc[cat1, cat2][1]
+                    pred_sum = matrix_err_counts.loc[cat1, cat2][2]
```

Comment:
I'm not sure if it is better to do it this way or just to pass the tp/tn/fn/fp (always?) in the matrix and then have the UI use that for calculating the metrics

Reply:
We've dealt with this in AutoML and made the choice to keep all 4 values for ease of use. You can get away with storing very few values if you store aggregates as well, but it just makes it harder to reason about, especially if users have direct access to the data.

Reply:
done, store all tp, fp, fn, tn now